clang/lib/Sema/SemaConcept.cpp

```diff
@@ -1144,6 +1164,46 @@ bool Sema::CheckInstantiatedFunctionTemplateConstraints(
                                      PointOfInstantiation, Satisfaction);
 }
 
+bool Sema::CheckFunctionConstraintsWithoutInstantiation(
```

Comment:
This is very similar to `CheckInstantiatedFunctionTemplateConstraints` can you either reuse that or share some code?

Reply:
I moved it into CheckFunctionTemplateConstraints() - calls to e.g. SetupConstraintCheckingTemplateArgumentsAndScope() inside couldn't be reused very simply so I added another internal function to handle MLTALs.